ip-utils.c

```diff
@@ -191,10 +223,32 @@ int janus_network_address_string_buffer_is_null(const janus_network_address_stri
 	return !b || b->family == AF_UNSPEC;
 }
 
-const char * janus_network_address_string_from_buffer(const janus_network_address_string_buffer *b) {
+const char *janus_network_address_string_from_buffer(const janus_network_address_string_buffer *b) {
 	if(janus_network_address_string_buffer_is_null(b)) {
 		return NULL;
 	} else {
 		return b->family == AF_INET ? b->ipv4 : b->ipv6;
 	}
 }
+
+int janus_network_lookup_interface(const struct ifaddrs *ifas, const char *iface, janus_network_address *result) {
+	if(result == NULL)
+		return -1;
+	janus_network_address_nullify(result);
+	if(iface == NULL)
+		return -1;
+	janus_network_query_config q;
+	/* Let's see if iface is an IPv4 address, an IPv6 address, or possibly an interface name */
+	if(janus_network_prepare_device_query(iface,
+			janus_network_query_options_ipv4 | janus_network_query_options_ipv6 | janus_network_query_options_name, &q)) {
+		/* None of them..? */
+		return -1;
```

Comment:
This should be `return -EINVAL` so the error type (invalid input arguments) is signaled as well as the bare fact that it is an error.